### DISCUSSION TITLE
Great job!

### DISCUSSION THREAD

**Discussion by tadejsv** [livingbio/typed-ffmpeg#356] [General]
Just used this library to replace `ffmpeg-python` - really easy migration, and the type hints and docstrings show up in VSCode without a hitch

**Comment by lucemia** [comment]
That's fantastic to hear! I'm thrilled you found the migration smooth and that the type hints and docstrings integrated seamlessly with VSCode. If you have any further feedback or questions as you continue to use the library, feel free to reach out. Your input helps us improve the project for everyone!

**Comment by hostage** [comment]
I also used this library to replace ffmpeg-python (after getting stuck with the bug where multiple upstream nodes have the same label) and it works really well for my use case (many concat filters -> xstack filter). The API design is clean and consistent.

(I don't have much experience with ffmpeg, so it would have been nice to get a few more examples of usage)

**Reply by lucemia** [reply]
Thanks! My own use case is simple, so I haven’t needed too many advanced examples. If you have specific types of examples in mind or a link to something that would have helped you, I’d love to hear them—happy to add more examples!

**Reply by hostage** [reply]
One thing was custom/unsupported filters; I needed the _color_ filter ([ffmpeg docs](https://ffmpeg.org/ffmpeg-filters.html#allrgb_002c-allyuv_002c-color_002c-colorchart_002c-colorspectrum_002c-haldclutsrc_002c-nullsrc_002c-pal75bars_002c-pal100bars_002c-rgbtestsrc_002c-smptebars_002c-smptehdbars_002c-testsrc_002c-testsrc2_002c-yuvtestsrc)) as sometimes when I am preparing a video with the xstack filter one of the inputs is just blank.

I first double and triple checked that the filter was not supported directly in the library (color got a lot of hits as a keyword, but not as a filter), then I figured out to call vfilter and (in this case) set an empty input stream type. It doesn't look like much when its finished, but having a practical example of calling vfilter (even for an already supported filter) would have saved me about 20 minutes of puzzling.

```python
windows[i] = ffmpeg.vfilter(
                    name="color",
                    color="black",
                    size=f"{sw}x{sh}",
                    duration=1,
                    input_typings=(),
                )
```

**Reply by lucemia** [reply]
> One thing was custom/unsupported filters; I needed the _color_ filter ([ffmpeg docs](https://ffmpeg.org/ffmpeg-filters.html#allrgb_002c-allyuv_002c-color_002c-colorchart_002c-colorspectrum_002c-haldclutsrc_002c-nullsrc_002c-pal75bars_002c-pal100bars_002c-rgbtestsrc_002c-smptebars_002c-smptehdbars_002c-testsrc_002c-testsrc2_002c-yuvtestsrc)) as sometimes when I am preparing a video with the xstack filter one of the inputs is just blank.
> 
> I first double and triple checked that the filter was not supported directly in the library (color got a lot of hits as a keyword, but not as a filter), then I figured out to call vfilter and (in this case) set an empty input stream type. It doesn't look like much when its finished, but having a practical example of calling vfilter (even for an already supported filter) would have saved me about 20 minutes of puzzling.
> 
> ```python
> windows[i] = ffmpeg.vfilter(
>                     name="color",
>                     color="black",
>                     size=f"{sw}x{sh}",
>                     duration=1,
>                     input_typings=(),
>                 )
> ```


Wow, thanks for your feedback! That’s super helpful—I forgot that I haven’t implemented the source filter yet. I’ll work on it sometime soon.

**Reply by lucemia** [reply]
@hostage
The new 2.6.3 release adds source filter support! 🎉 Now you can do:

```
import ffmpeg

# Generate a red HD video
ffmpeg.sources.color(color="red", size="hd1080").output(filename="output.mp4").compile()

# Use a color source as an input alongside another file
ffmpeg.output(
     in_file, ffmpeg.sources.color(color="red", size="hd1080"), filename="output.mp4"
).compile()
```
Hope you enjoy the update! 🚀

**Reply by hostage** [reply]
Thanks for maintaining and improving the project, I think it is very useful!

**Comment by jasonholder1122** [comment]
I switched to this library as an alternative to ffmpeg-[python ](https://www.jobz.pk/it-employment/)after encountering issues with duplicate labels in complex filter graphs (many concat filters feeding into an xstack filter). It has proven to be a solid solution for my workflow, thanks to its well-designed and consistent API.

Although I’m not very experienced with FFmpeg, having more usage examples would have been helpful for getting started more smoothly.

**Reply by lucemia** [reply]
@jasonholder1122 Thanks for the kind words! Glad to hear it’s working well for your workflow. We’ll definitely keep working on adding more usage examples to make getting started smoother.

**Reply by hostage** [reply]
That account is using an LLM to repost comments. Very lame.
